Support for description in imgtable

Grtschnk opened the pull request:
## Description:

Edit: Oh boy, not sure if I got the branch selection right? Not intending to commit the whole backlog, just my last two commits.

Python function ImageTableDirective for imgtable extended to allow further optional arguments. Arguments merged into description string, displayed underneath link/name. As of now, it creates an additional paragraph inside the table cell, might take up too much space. The alternative would be appending the text with a newline (or adjusting CSS). Feel free to adjust or let me know.

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/1136

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** n/a<esphome PR number goes here>

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`. Using 'next' branch as it is a new feature (and design adjustment) for the documentation.
  - [ ] **n/a** Link added in `/index.rst` when creating new documents for new components or cookbook.

Grtschnk committed:
Optional description argument added to imgtable

Optional 4th argument in imgtable to be used as description under the name/link. 
(actually unlimited, additional commas/arguments get merged, in case description contains commas)

Grtschnk committed:
Example descriptions for a few sensors

Minimal so far. Showcases use of additional comma for Dallas sensors

frenck reviewed (changes requested):
Hi there @Grtschnk 👋 

Fixed the target branch, so it looks better now 😉 

I don't see where this change should so up?
Also, there is a merge conflict, could you take a look at resolving that?

Thanks! 👍 

../Frenck

Grtschnk committed:
Merge branch 'current' into description_imgtable

frenck commented:
The reason I ask: If I look at the build of this branch:

![image](https://user-images.githubusercontent.com/195327/117645418-dd387500-b18a-11eb-928d-6ce5ad760839.png)

I would have expected to see something 😬 

See for yourself: <https://deploy-preview-1016--esphome.netlify.app/#sensor-components>

Grtschnk committed:
fixed description_imgtable

Fixed missing line to fill variable from buffered list.
Fixed comparison operator

Grtschnk commented:
Of course, it wouldn't work. 
My earlier commit missed lines I had locally. Fixed it.

Grtschnk committed:
Merge branch 'current' into description_imgtable

ludeeus commented on github.py:
```diff
@@ -111,15 +111,19 @@ class ImageTableDirective(Table):
 
     def run(self):
-        cols = self.options.get("columns", 3)
+        cols = self.options.get("columns", 4)
```
IMO 3 is already pushing it for mobile, until there can be a different amount of columns for desktop/mobile I don't think this should change.

frenck replied:
```suggestion
        cols = self.options.get("columns", 3)
```

Grtschnk committed:
Update github.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Grtschnk committed:
Merge branch 'current' into description_imgtable
